src/librustc_typeck/check/mod.rs

```diff
@@ -1014,11 +1014,17 @@ fn check_impl_items_against_trait<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
                                            trait_const,
                                            &impl_trait_ref);
                     } else {
-                        span_err!(tcx.sess, impl_item.span, E0323,
+                         struct_span_err!(tcx.sess, impl_item.span, E0323,
                                   "item `{}` is an associated const, \
                                   which doesn't match its trait `{:?}`",
                                   impl_const.name,
                                   impl_trait_ref)
+                         .span_label(impl_item.span, &format!("does not match trait"))
+                         .span_label(
+                            tcx.map.span_if_local(ty_trait_item.def_id()).unwrap(),
```

Comment:
btw, I asked in #rust-internals about using span_if_local here, and got: "span_if_local is not okay here; trait defn may come from an extern crate and it will panic then"

You may want to try another direction